bin/main.ml

```diff
@@ -0,0 +1,148 @@
+(* let absolute_path path =
+  let cwd = Sys.getcwd ()
+  in Filename.concat cwd path *)
+
+let getenv name =
+  try
+    Some (Sys.getenv name)
+  with Not_found -> None
+
+let flat_map (f: 'a -> 'b list) (a : 'a list) : 'b list =
+  a
+  |> List.map f
+  |> List.flatten
+
+let safe_list_hd lst = 
+  try
+    Some (List.hd lst)
+  with Failure _ -> None
+
+let rec find_parent_directory_with_file target_file start_dir =
+  let target_file = Filename.concat start_dir target_file
+  in
+    if Sys.is_directory target_file then
+      start_dir
+    else
+      (* Move one dir up and try again *)
+      Filename.dirname start_dir
+      |> find_parent_directory_with_file target_file
+
+let parent_dir () =
+  let start = Sys.getcwd ()
+  (* TODO: make env variable with default *)
+  in let target = "issue"
+  in find_parent_directory_with_file target start
+
+(* TODO: use path handler *)
+let issue_dir () =
+  let dir = parent_dir ()
+  in Filename.concat dir "issue"
+
+let rec traverse_directory path =
+  if Sys.is_directory path then
+    Sys.readdir path
+    |> Array.to_list
+    |> flat_map(fun file ->
+        Filename.concat path file
+        |> traverse_directory
+      )
+  else [path]
+
+let wrap_in_article issue_html =
+  "<article>" ^ issue_html ^ "</article>"
+
+let read_entire_file path =
+  In_channel.with_open_text path In_channel.input_all
+
+let lines_of_file path =
+  In_channel.with_open_text path In_channel.input_lines
+
+let file_to_html file =
+  read_entire_file file
+    |> Omd.of_string
+    |> Omd.to_html
+
+let to_html root =
+  traverse_directory root
+    |> List.map (fun x -> x |> file_to_html |> wrap_in_article)
+    |> String.concat "\n\n\n"
+
+let list root =
+  traverse_directory root
+  |> List.filter(fun file ->
+      file |> Filename.extension |> String.lowercase_ascii |> String.equal ".md"
+    )
+  |> List.iter(fun file -> print_endline file)
+
+let safe_filename filename = 
+  let r = Str.regexp "[^A-Za-z0-9.-]" in
+  Str.global_replace r "_" filename
+
+(* FIXME: isn't cross platform compatible, Filename.dir_sep is a string
+   so I can't use String.split_on_char *)
+let path_split = String.split_on_char '/'
+
+let rec mkdir_p path = 
+  if not (Sys.file_exists path) then begin
+    let parent = Filename.dirname path in
+    if path <> parent then begin
+      mkdir_p parent
+    end;
+    Sys.mkdir path 0o777
+  end
+
+let move from to' = Sys.rename from to'
+
+let open_file_with_editor path =
+  let editor = Option.value ~default:"vi" (getenv "EDITOR") in
+  (* open the temporary file with the default editor *)
+  Printf.sprintf "%s %s" editor path
```

Comment:
The second `%s` needs quoting too.

I think `"%s '%s'"` (single quote) would be most appropriate as no variable interpolation is needed or desired.

Another edge case that might exist is that the values passed to sprintf might have single quotes.

I think we might need a more high level api that is built for spawning and passing options to commands.